acceptance/internal/prepare_server.go

```diff
@@ -90,6 +90,9 @@ func startDedicatedServer(t *testing.T,
 		s.RequestCallback = func(request *testserver.Request) {
 			req := getLoggedRequest(request, includeHeaders)
 			reqJson, err := json.MarshalIndent(req, "", "  ")
+
+			defer s.LockUnlock()()
```

Comment:
nit: Can move this to the top of this function?

Reply:
It's on purpose, no need to lock the above two lines, they don't operate on shared data.